docs/adr/0007-limit-visits-migration-to-10000.md

```diff
@@ -0,0 +1,163 @@
+  - 50th percentile: 400 ms
+  - 75th percentile: 2,500 ms (2.5 seconds)
+  - 90th percentile: 6,400 ms (6.4 seconds)
+  - 95th percentile: 11,000 ms (11 seconds)
+  - 99th percentile: 25,000 ms (25 seconds)
+  - 99.9th  percentile: 50,000 ms (50 seconds)
+
+### Problem Statement
+Given the observations from the dry-run experiment, the rest of the document examines an approach to answer the question: **How can we increase the rate of which migrations end, and simultaneously keep the user’s database size at a reasonable size?**
+
+The user implication of keeping the rate of ended migrations high is that users keep their history, and can interact normally with the URL bar to search history, searching history in the history panel and navigating to sites they visited in the past.
+
+
+
+The user implication of keeping a reasonable database size is that the database is less likely to lock on long queries. Meaning we reduce performance issues when users use the search bar, the history panel and when navigating to sites.
+
+Finally, it’s important to note that power users and daily active users will be more likely to have large histories and thus:
+- Power users are more likely to fail their migration.
+- Power users are more likely to have performance issues with history and the search bar.
+  - We saw a version of this with Favicons in an earlier incident, where users were coming across a significant number of database locks, crashing the app. This isn’t to say that the incident is directly related to this, however, large histories could contribute to the state we saw in the incident as it would take longer to run the queries.
+
+## Decision Drivers <!-- optional -->
+
+* We must not lose users’ recent history.
+  * What is considered “recent” is not concretely defined. There is prior art, however:
+    * [Firefox Sync in Android syncs 5000 visits.](https://github.com/mozilla/application-services/blob/c34de065f5773b529a0def773cffb29b45676bce/components/places/src/history_sync/mod.rs#L16)
+    * [Firefox Desktop and Android impose limits on the size of the database, 75 MiB in Android](https://github.com/mozilla-mobile/firefox-android/blob/8077db4b55c1a8976ec130462767e3075b9cbeab/android-components/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageWorker.kt#L38)
+    * [When importing history from chrome to Firefox, we only migrate the latest 2000 visits](https://searchfox.org/mozilla-central/rev/f40d29a11f2eb4685256b59934e637012ea6fb78/browser/app/profile/firefox.js#2134)
+    * [Chrome only keeps a user’s history for 90 days](https://support.google.com/chrome/answer/95589)
+
+* User’s experience with History must not regress, and ideally should improve.
+  * User experience is tightly coupled with the size of the database. The larger the database, the longer queries take. The longer queries take, the longer it would take for a user to observe their searched history and the history panel.
+
```

Comment:
It might be worth talking about the maintenance process we now have, which also has a "target size" for the db?